linkis-computation-governance/linkis-engineconn-manager/linkis-engineconn-manager-core/src/main/scala/com/webank/wedatasphere/linkis/ecm/core/launch/ProcessEngineConnLaunch.scala

```diff
@@ -130,6 +131,14 @@ trait ProcessEngineConnLaunch extends EngineConnLaunch with Logging {
     var springConf = Map("spring.application.name" -> GovernanceCommonConf.ENGINE_CONN_SPRING_NAME.getValue,
       "server.port" -> engineConnPort, "spring.profiles.active" -> "engineconn",
       "logging.config" -> s"classpath:${EnvConfiguration.LOG4J2_XML_FILE.getValue}") ++: discoveryMsgGenerator.generate(engineConnManagerEnv)
+
+    //暂时通过判断engineConnManagerHost是否为IP地址来判断是否使用eureka.instance.prefer-ip-address，后续需要通过engineConnManagerEnv.properties获取相关属性
+    logger.info(s"engineConnManagerHost:" + engineConnManagerEnv.engineConnManagerHost)
```

Comment:
It is recommended to read Commvars to read this Key（EUREKA_PREFER_IP）, which will contain the key Value of the environment variable，CommonVars[String]("EUREKA_PREFER_IP", false)，If there is a value in the environment variable, it will not take the default false

Reply:
thanks, I will try and submit changes.

Reply:
I have updated，pls review.

Reply:
Thank you, you should use the environment variable EUREKA_PREFER_IP instead of spring parameters, because Commvar will read the environment variables

Reply:
Hello, to facilitate communication, you can add my wechat

Reply:
sorry,I can't find you webchat, I send my webchat to your email:wpeace1212@gmail.com,pls check it.

Reply:
in linkis-env.sh export EUREKA_PREFER_IP, in Configuration read it through CommonVars, and related code has been modified.